draft-ietf-httpbis-safe-method-w-body.xml

```diff
@@ -346,6 +367,15 @@ Dubois, Camille, camille.dubois@example.net
       considerations as all HTTP methods as described in
       <xref target="HTTP"/>.
     </t>
+
+    <t>
+      The QUERY method can be used as an alternative to passing query
+      information in the query portion of a URI. This is preferred in some
+      cases, as the URI is more likely to be logged than the request content. If
+      a server creates a temporary resource to represent the results of a QUERY
+      request (e.g. for use in the Location or Content-Location field), the URI of this
+      resource &SHOULD-NOT; expose the original request content in plaintext.
```

Comment:
I'm not sure that this is a consideration that warrants a "SHOULD NOT".  Yes, there is a risk that logging a URL will cause stuff to be logged that you shouldn't log, but that is only if the QUERY contains stuff that shouldn't be logged.  This reads as being unconditional.

Reply:
I agree.

Reply:
maybe... "... the URI SHOULD be chosen in a way that potentially sensitive information is not exposed."?

Reply:
I'm looking for an additional qualification on the statement, something like:

> If a server creates a temporary resource to represent the results of a QUERY request (e.g., for use in the Location or Content-Location field), the URI of this resource &SHOULD-NOT; expose **any sensitive information from** the original request content in plaintext.